addon/ember-test-waiters/wait-for.ts

```diff
@@ -71,11 +64,35 @@ export default function waitFor(
 
     let fn = descriptor.value;
 
-    descriptor.value = function(...args: any[]) {
-      let promise = fn.call(this, ...args);
-      return waitForPromise(promise, label);
-    };
+    descriptor.value = wrapFunction(fn, label);
 
     return descriptor;
   }
 }
+
+function wrapFunction(fn: Function, label?: string) {
+  if (!DEBUG) {
+    return fn;
+  }
+
+  return function(this: any, ...args: any[]) {
+    let result = fn.call(this, ...args);
+
+    if (isThenable(result)) {
+      return waitForPromise(result, label);
+    } else {
+      return result;
+    }
```

Comment:
Note: @bendemboski I think we would add the check for iterator here just like the `isThenable` thing (iterator being something with a `typeof result.next === 'function'` I believe)

Reply:
Ya maybe add an `isIterable`?